### DISCUSSION TITLE
Magick compose & compos

### DISCUSSION THREAD

**Discussion by Wiggy1981** [ImageMagick/ImageMagick#5425] [Digital Image Processing]
I have a problem -  i create 2 text titles, then save them as separate images i add them to a existing background. This all works fine when the font fill is white, but when i use another color it's like I'm only getting the alpha channel when the final image is compositied. The individual files look fine though.

```
magick -background none -fill 'white' \ 
                           -font Fagrak.otf -pointsize 72 label:"Power Matrix Method" \ 
                                                         pmm.jpg 

magick -background none -fill 'gold' \
          -font Fagrak.otf -pointsize 72 label:"CANDIDATE FEEDBACK" \ 
                                                          cfb.jpg

magick composite -gravity North -geometry +0+20 pmm.jpg logo_black.jpg cover_black.jpg
 
magick composite -gravity South -geometry +0+0 cfb.jpg cover_black.jpg cover_blackf.jpg
magick composite -gravity South -geometry +0+200 marknew.jpg cover_blackf.jpg cover_blackff.jpg
```

**Comment by snibgo** [comment]
You don't show all the inputs, nor the outputs, so diagnosis needs guesswork.

Your pmm.jpg is grayscale, so will have only one channel when IM reads it. If your outputs are grayscale when you expect colour, that is probably the problem. See http://www.imagemagick.org/script/porting.php .

A fix is to insert `-set colorspace sRGB` after reading inputs that may be grayscale.

As a general point, I do not recommend using `magick composite`. Instead, use just `magick` with the `-composite` operation, and reverse the order of the two input files

**Reply by Wiggy1981** [reply]
I can't get the -set colorspace sRGB to work with composite. 

magick composite -set colorspace sRGB -gravity North -geometry +0+20 -set colorspace sRGB pmm.jpg logo_black.jpg cover_b>
magick composite -set colorspace sRGB -gravity South -geometry +0+0 -set colorspace sRGB cfb.jpg cover_black.jpg cover_b>
magick composite -set colorspace sRGB -gravity South -geometry +0+200 -set colorspace sRGB marknew.jpg cover_blackf.jpg >

I get the error 

composite: unrecognized option `-set' @ error/composite.c/CompositeImageCommand/1456.                                 composite: unrecognized option `-set' @ error/composite.c/CompositeImageCommand/1456.                                 composite: unrecognized option `-set' @ error/composite.c/CompositeImageCommand/1456.

Is there some way to stop imagemagick automatically choosing grey scale for operations?

**Comment by snibgo** [comment]
I suggest:
```
magick logo_black.jpg -set colorspace sRGB pmm.jpg -gravity North -geometry +0+20 -composite cover_black.jpg
 
magick cover_black.jpg -set colorspace cfb.jpg -gravity South -geometry +0+0 -composite cover_blackf.jpg

magick cover_blackf.jpg -set colorspace marknew.jpg -gravity South -geometry +0+200 -composite cover_blackff.jpg
```
EDIT: "`magick composite`" is provided for backwards compatibility (ie very old scripts). I suggest it should never be used for new work. Instead, use just "`magick`" with the "`-composite`" operation.

**Reply by Wiggy1981** [reply]
Hi thanks for the input. When I updated the script I now get this error...

`magick: image sequence is required `-composite' @ error/operation.c/CLIListOperatorImages/4026.
magick: image sequence is required `-composite' @ error/operation.c/CLIListOperatorImages/4026.
`

With the following updated script.

`magick  -background black -fill 'green1' \
          -font Fagrak.otf -pointsize 72 label:"Power Matrix Method" \
          pmm.jpg

  magick -background black -fill 'gold' \
          -font Fagrak.otf -pointsize 72 label:"CANDIDATE FEEDBACK" \
          cfb.jpg

magick logo_black.jpg -set colorspace sRGB pmm.jpg -gravity North -geometry +0+20 -composite cover_black.jpg

magick cover_black.jpg -set colorspace cfb.jpg -gravity South -geometry +0+0 -composite cover_blackf.jpg

mqgick cover_blackf.jpg -set colorspace marknew.jpg -gravity South -geometry +0+200 -composite cover_blackff.jpg`

**Comment by snibgo** [comment]
Which command gives that error?

What version of IM do you use? On what platform (bash, Windows BAT, or whatever)?

Your first two commands each seem to be split across three lines. Perhaps that is just a formatting problem on the forum. In a script, if you split a command, you need to use line-continuation characters: backslash \\ on bash, or caret ^ on Windows.

**Reply by Wiggy1981** [reply]
I am using ImageMagick 7.1.0-29, I'm using Termux.  the commands are one one line each on my script. I think it's a formatting issue with the forum post.

When I ran the bash script containing the commands. That was the error a got back.

**Comment by snibgo** [comment]
> When I ran the bash script containing the commands. That was the error a got back.

Yes, but there are five commands in your script. There is one error message. Which command created the error message?

I don't know "Termux". If that doesn't tell you which command created the eror, then try running a script with just the first command. If that works then add the second command and run it again. And so on.
